client/html/themes/aimeos.js

```diff
@@ -1225,14 +1225,28 @@ AimeosCatalog = {
 					if( parent.data("reqstock") && $(".stockitem", newStock).hasClass("stock-out") ) {
 						$(".addbasket .btn-action", parent).addClass("btn-disabled").attr("disabled", "disabled");
 					} else {
-						$(".addbasket .btn-action", parent).removeClass("btn-disabled").removeAttr("disabled");
+                        if(AimeosCatalog.validateVariant()) {
+    						$(".addbasket .btn-action", parent).removeClass("btn-disabled").removeAttr("disabled");
+                        }
 					}
 
 					$(".catalog-detail-additional .subproduct-actual").removeClass("subproduct-actual");
 					$(".catalog-detail-additional .subproduct-" + prodId).addClass("subproduct-actual");
 				}
 			}
+
+            if(!AimeosCatalog.validateVariant()) {
+                $(".addbasket .btn-action", parent).addClass("btn-disabled").attr("disabled", "disabled");
+            }
+
+            if(!AimeosCatalog.validateVariant()) {
+                $(".articleitem").removeClass("stock-actual");
+            }
 		});
+
+        if(!AimeosCatalog.validateVariant()) {
+            $(".articleitem").removeClass("stock-actual");
+        }
```

Comment:
Why are these lines necessary? They are executed on page load

Reply:
These lines is to hide the message if no attribute selected.
![image](https://user-images.githubusercontent.com/82187386/120067161-c495ed80-c0ac-11eb-8cba-8839b649684d.png)

Hope it make sense.
Please guide me to do proper way.

Reply:
Maybe it's better to remove the `stock-actual` class by default for `select` and `group` products in the template:
https://github.com/aimeos/ai-client-html/blob/master/client/html/templates/catalog/detail/body-standard.php#L270

Otherwise, customer will notice that the line disappears because the JS is executed after the page is loaded. Thus, we would suggest something like:

```
<div class="articleitem <?= !in_array( $this->detailProductItem->getType(), ['select', 'group'] ) ? 'stock-actual' : '' ?>"
```
The same should be done here:
- https://github.com/aimeos/ai-client-html/blob/master/client/html/templates/catalog/lists/items-body-list.php#L96
- https://github.com/aimeos/ai-client-html/blob/master/client/html/templates/common/partials/products-standard.php#L238